src/sympc/session/session.py

```diff
@@ -186,44 +186,53 @@ def get_protocol(self) -> Protocol:
         return self.protocol
 
     def _generate_random_share(
-        self,
-        shape: Union[tuple, torch.Size],
+        self, shape: Union[tuple, torch.Size], ring_size: int
     ) -> List[torch.Tensor]:
-        """Generate random tensor share for the given shape.
+        """Generate random tensor share for the given shape and ring_size.
 
         Args:
             shape (Union[tuple, torch.Size]): Shape for the share.
+            ring_size(int): ring size to generate share.
 
         The generators are invoked in Counter(CTR) mode as parties with the same initial seeds
-        could generate the same random number on subsequent invocations.
+        could generate correlated random numbers on subsequent invocations.
 
         Returns:
             List[torch.Tensor] : shares generated by the generators.
         """
+        from sympc.tensor import PRIME_NUMBER
+
         gen1, gen2 = self.przs_generators
+        tensor_type = get_type_from_ring(ring_size)
 
         tensor_share1 = generate_random_element(
-            tensor_type=self.tensor_type,
+            tensor_type=tensor_type,
             generator=gen1,
             shape=shape,
         )
 
         tensor_share2 = generate_random_element(
-            tensor_type=self.tensor_type,
+            tensor_type=tensor_type,
             generator=gen2,
             shape=shape,
         )
 
-        return [tensor_share1, tensor_share2]
+        # PRIME_NUMBER ring share is generated in torch.uint8 datatype
+        if ring_size == PRIME_NUMBER:
```

Comment:
Q: Shouldn't we take into consideration also the case when `ring_size` is 2

Reply:
For binary ring size the tensor type maps ---> torch.bool,so only binary values are generated, for prime rings tensor type maps--->torch.uint8 `[0-255]` ,use modulus to scale to `[0,67]`